lib/x.h

```diff
@@ -0,0 +1,43 @@
+// SPDX-FileCopyrightText: 2023-2025, Alejandro Colomar <alx@kernel.org>
+// SPDX-License-Identifier: BSD-3-Clause
+
+
+#ifndef SHADOW_INCLUDE_LIB_X_H_
+#define SHADOW_INCLUDE_LIB_X_H_
+
+
+#include "config.h"
+
+#include <errno.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "shadowlog.h"
+
+
+#define x(p)                                                          \
```

Comment:
The name x() might be too generic and could conflict with X macro patterns commonly used in C (as acknowledged in revision v5). Consider a more descriptive name like exit_on_null() or ensure_nonnull().